products/bridge/smart-contracts/contracts/core/Relayer.sol

```diff
@@ -28,10 +24,21 @@ contract Relayer is IRelayer {
     function relay(
         uint targetChainId,
         address target,
-        bytes calldata call,
+        bytes4 callSelector,
+        bytes calldata callData,
         uint gasLimit
     ) external returns (uint) {
-        emit Relayed(targetChainId, target, call, gasLimit, nonce);
+        emit Relayed(
+            targetChainId,
+            target,
+            abi.encodeWithSelector(
+                callSelector,
+                abi.encode(CallMetadata(block.chainid, msg.sender)),
+                callData
+            ),
```

Comment:
Relay now appends chainid and sender as metaparameters for the target to use this information